src/robotFactory.js

```diff
@@ -1,10 +1,84 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    if (step > 0) {
+      this.coords.y += step;
+    }
+  }
+
+  goBack(step = 1) {
+    if (step > 0) {
+      this.coords.y -= step;
+    }
+  }
+
+  goRight(step = 1) {
+    if (step > 0) {
+      this.coords.x += step;
+    }
+  }
+
+  goLeft(step = 1) {
+    if (step > 0) {
+      this.coords.x -= step;
+    }
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    if (step > 0) {
+      this.coords.z += step;
+    }
+  }
+
+  goDown(step = 1) {
+    if (step > 0) {
+      this.coords.z -= step;
+    }
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipversion, maxLoadWeight,
+    currentLoad = null) {
+    super(name, weight, coords, chipversion);
```

Comment:
You should keep all the args in one line or each in a separate one
```suggestion
class DeliveryDrone extends FlyingRobot {
  constructor(
    name, 
    weight, 
    coords, 
    chipversion, 
    maxLoadWeight,
    currentLoad = null,
  ) {
    super(name, weight, coords, chipversion);
```